var/spack/repos/builtin/packages/fairlogger/package.py

```diff
@@ -71,20 +118,18 @@ def patch(self):
         if not self.spec.satisfies("@develop"):
             filter_file(
                 r"(get_git_version\(.*)\)",
-                r"\1 DEFAULT_VERSION %s)" % self.spec.version,
+                rf"\1 DEFAULT_VERSION {self.spec.version})",
                 "CMakeLists.txt",
             )
 
     def cmake_args(self):
-        args = []
-        args.append("-DDISABLE_COLOR=ON")
-        cxxstd = self.spec.variants["cxxstd"].value
-        if cxxstd != "default":
-            args.append("-DCMAKE_CXX_STANDARD=%s" % cxxstd)
+        args = [self.define("DISABLE_COLOR", True)]
+        if self.spec.variants["cxxstd"].value != "default":
+            args.append(self.define_from_variant("CMAKE_CXX_STANDARD", "cxxstd"))
         if self.spec.satisfies("@1.4:"):
             args.append(self.define_from_variant("USE_BOOST_PRETTY_FUNCTION", "pretty"))
         if self.spec.satisfies("@1.6:"):
-            args.append("-DUSE_EXTERNAL_FMT=ON")
+            args.append(self.define("USE_EXTERNAL_FMT", True))
         if self.spec.satisfies("^boost@:1.69"):
-            args.append("-DBoost_NO_BOOST_CMAKE=ON")
+            args.append(self.define("Boost_NO_BOOST_CMAKE", True))
```

Comment:
Just FYI, I am waiting for somebody to review https://github.com/spack/spack/pull/36198 at which point this shouldn't be necessary.

Reply:
ah, cool. Can I be of help with the review of #36198? Not yet super familiar with the review process here, in case I can do anything to help, let me know!